examples/website/src/shared/route/baseTemplateContainer.tsx

```diff
@@ -64,16 +74,35 @@ export function BaseTemplateContainer({
   )
 }
 
-function dataToNavigation(data: any): NavigationItem[] {
-  return data.links
-    .filter((link: {__typename: string}) => link.__typename == 'PageNavigationLink')
-    .map((link: any) => linkToNavigationItem(link))
-}
-
 function linkToNavigationItem(link: any): NavigationItem {
-  return {
-    title: link.label,
-    route: PageRoute.create({slug: link.page?.slug ?? undefined}),
-    isActive: false
+  switch (link.__typename) {
+    case 'PageNavigationLink':
+      return {
+        title: link.label,
+        route: PageRoute.create({
+          slug: link.page?.slug ?? undefined
```

Comment:
Here is an example where you should use `!`. TypeScript doesn't know that when `link.__typename` is of type `PageNavigationLink` we have a page for sure. But we developers do. You can rewrite this line to `slug: link.page!.slug`.